vault/token_store.go

```diff
@@ -1169,6 +1169,39 @@ func (ts *TokenStore) revokeSalted(ctx context.Context, saltedId string) (ret er
 		}
 	}
 
+	// Mark all children token as orphan by removing
+	// their parent index, and clear the parent entry.
+	//
+	// Marking the token as orphan is the correct behavior in here since
+	// revokeTreeSalted will ensure that they are deleted anyways if it's not an
+	// explicit call to orphan the child tokens (the delete occurs at the leaf
+	// node and uses parent prefix, not entry.Parent, to build the tree for
+	// traversal).
+	parentPath := parentPrefix + saltedId + "/"
+	children, err := ts.view.List(ctx, parentPath)
+	if err != nil {
+		return fmt.Errorf("failed to scan for children: %v", err)
+	}
+	for _, child := range children {
+		entry, err := ts.lookupSalted(ctx, child, true)
+		if err != nil {
+			return fmt.Errorf("failed to get child token: %v", err)
+		}
+		lock := locksutil.LockForKey(ts.tokenLocks, entry.ID)
```

Comment:
If this lock happens to be the same lock that is being used outside of this loop, acquiring this will deadlock. No?

Reply:
Never mind.